src/strings_utils.cpp

```diff
@@ -235,10 +235,12 @@ bool is_ascii_normal(const std::string str){
 	auto it = find_if(str.begin(), str.end(), [](int x) {
 		// The behavior is undefined if the value of ch is not representable as unsigned char and is not equal to EOF.
 		// http://en.cppreference.com/w/cpp/string/byte/isprint
-		if ((x < std::numeric_limits<unsigned char>::min()
-			|| x > std::numeric_limits<unsigned char>::max())
-			&& x != EOF)
-				return true;
+		bool in_range =
+			(x >= std::numeric_limits<unsigned char>::min()) ||
+			(x <= std::numeric_limits<unsigned char>::max());
+		bool valid = in_range && ( x!= EOF );
```

Comment:
"The behavior is undefined if the value of ch is not representable as unsigned char and is not equal to EOF."
if EOF then it is fine  (even if not in range)

But! EOF is guaranteed to be <0,  e.g.  -1,  so actually    if  in_range  is true then x!=EOF
http://en.cppreference.com/w/c/io  " EOF | integer constant expression of type int and negative value"

static_assert(  (EOF)  < 0  ,  "EOF must be negative value");  // http://en.cppreference.com/w/c/io
bool valid == in_range; // no need to test regarding EOF, as EOF is < 0